src/CLI.cpp

```diff
@@ -0,0 +1,197 @@
+#include <fstream>
+#include <functional>
+#include <iostream>
+#include <limits>
+#include <map>
+#include <sstream>
+#include <string>
+
+#include "CLI.h"
+#include "Error.h"
+#include "Param.h"
+
+void parse_read_file(std::string const& input, std::string& output)
+{
+	// check to see if the file exists and error out if it doesn't
+	if (static_cast<bool>(std::ifstream(input)) == false)
+	{
+		ERR_CRITICAL_FMT("%s is not a file", input.c_str());
+	}
+	output = input;
+}
+
+void parse_write_dir(std::string const& input, std::string& output)
+{
+	// check to see if this prefix already exists as a file and error out
+	if (static_cast<bool>(std::ifstream(input)) == true)
+	{
+		ERR_CRITICAL_FMT("Cannot use this prefix, this path already exists"
+						 " as a file: %s", input.c_str());
+	}
+	// TODO: add a platform-independent check to see if the prefix could
+	// be added as a directory or file
+	output = input;
+}
+
+void parse_integer(std::string const& input, int& output)
+{
+	try
+	{
+		output = std::stoi(input);
```

Comment:
Can we do something like:
```suggestion
std::size_t pos;
output = std::stoi(input, &pos);
if (pos != input.len)
{
  ERR_CRITICAL_FMT("Unable to completely parse integer: %s\n", input.c_str());
}
```

Similarly for float parsing later on.

Reply:
Done